packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -794,16 +800,15 @@ export class Relayer {
     // First try to fetch deposit from the L2 client's default block search config. This should work in most cases.
     let deposit: Deposit | undefined = this.l2Client.getDepositByHash(relay.depositHash);
     if (deposit !== undefined) return deposit;
-    // We could not find a deposit using the L2 client's default block search config. Next, we'll modify the block
-    // search config bridge pool's deployment time. This allows us to capture any deposits that happened outside of
-    // the L2 client's default block search config.
+    // We could not find a deposit using the L2 client's default block search config. As a fallback, we'll modify the
+    // the block search config such that we can find the valid deposit if it exists. This fallback logic is expensive
+    // and results in a lot of web3 requests, but will find the deposit in the worst case.
+    // TODO: We could improve this search by only looking +/- 10 minutes within the quote time, as this is a constraint
+    // enforced by the BridgeDepositBox. The hard part is that determining the average time per L2 block is not trivial.
     else {
-      const bridgePoolDeploymentTime = (
-        await this.l2BlockFinder.getBlockForTimestamp(this.deployTimestamps[relay.l1Token])
-      ).number;
       let blockSearchConfig = {
-        fromBlock: bridgePoolDeploymentTime,
-        toBlock: bridgePoolDeploymentTime + this.l2LookbackWindow,
+        fromBlock: 0,
```

Comment:
This is the one part I'm not happy about. This event querying seems expensive in the worst case. Arbitrum team did state that Infura won't have limited event block search configs soonTM so maybe this code can be deprecated shortly, but I'm not sure what the best way to optimize this search is. We could try to look +/- 10 minutes around the deposit.quoteTime but timing on L2 is a dangerous assumption to make.

Reply:
ye, and I feel that the call to `getCodeDeployedAtTimestamp` is also pretty slow.